explorer/interpreter/type_checker.cpp

```diff
@@ -826,6 +830,10 @@ auto TypeChecker::ExpectNonPlaceholderType(SourceLocation source_loc,
            << "invalid use of mixin "
            << mixin_type->mixin_type().declaration().name();
   }
+  if (auto* namespace_type = dyn_cast<TypeOfNamespaceName>(type)) {
+    return ProgramError(source_loc)
+           << "expected `.member_name` after name of " << *namespace_type;
+  }
```

Comment:
Note, I think this is somewhat what I'm expressing concern with above, with how namespaces are classified as types but then member access is _required_.